static/app/actionCreators/account.tsx

```diff
@@ -25,20 +26,23 @@ export async function disconnectIdentity(
   onSuccess();
 }
 
-export function updateUser(user: User) {
+export function updateUser(user: User | ChangeAvatarUser) {
   const previousUser = ConfigStore.get('user');
 
   // If the user changed their theme preferences, we should also update
   // the config store
   if (
+    user.options &&
     previousUser.options.theme !== user.options.theme &&
     user.options.theme !== 'system'
   ) {
     ConfigStore.set('theme', user.options.theme);
   }
 
+  const options = {...previousUser.options, ...user.options};
+
   // Ideally we'd fire an action but this is gonna get refactored soon anyway
-  ConfigStore.set('user', user);
+  ConfigStore.set('user', {...previousUser, ...user, options});
```

Comment:
Maybe explain here we are merging because the avatar endpoint doesn't return a full `User`